module/x/gravity/migrations/v3/store.go

```diff
@@ -0,0 +1,126 @@
+package v3
+
+import (
+	"encoding/hex"
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/Gravity-Bridge/Gravity-Bridge/module/x/gravity/types"
+)
+
+// MigrateStore performs in-place store migrations from v2 to v3. The migration
+// includes:
+//
+// - Migrate all attestations to their correct new keys
+func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec) error {
```

Comment:
how does this handle a potential collision? it should at least detect one and panic, I don' think full handling is required becuase it should not exist but I want to be very sure of course.